format old code and autoformat new code

louiechristie opened the pull request:
- [x] This PR follows [contributing quidelines](CONTRIBUTING.md)
- [ ] This PR has tests (for backend in particular)


### What change does this PR introduce?

- Autoformat code on save(HTML, SCSS, markdown, JavaScript) - VSCode settings changed to autoformat new code on save 
- Autoformat code on pre-commit (HTML, SCSS, markdown, JavaScript)
- All old code formatted

### Notes

* Open some code in VSCode
* Install the recommended extensions (Prettier and Beautify)
* Edit some code, HTML, SCSS, markdown or JavaScript.
* Save it - the code should autoformat nicely.

* Disable auto format in VSCode settings or use a different editor to edit some code and format it badly
* git commit 
* look at commit history using SourceTree, Git Graph or similar, the commit's code should have been autoformatted.

### Ticket

- [Trello Ticket](https://trello.com/c/KuFBknWk/122-autoformat-code-html-scss-markdown-javascript)

### Screenshots

https://user-images.githubusercontent.com/6807448/107781154-78ae7b00-6d3f-11eb-8186-17913a275bc4.mp4

Commit:
modified settings with own db url

Commit:
cafe view retrieving cafe from model by ID

Commit:
opened up spaces in view comments

Commit:
removed my local db setting in settings

chriswedgwood committed:
Update README.md

Database doc improvements

lili2311 committed:
Update README.md

lili2311 committed:
Merge pull request #35 from PolyglotDevsLondon/feat/venue-view-and-template

Feat/venue view and template

louiechristie committed:
Merge develop branch into master

louiechristie committed:
Merge branch 'develop'

louiechristie committed:
chore: autoformat code on save in vscode

louiechristie committed:
fix: dijango variable inside js inside html format

louiechristie committed:
chore: format code

louiechristie committed:
Merge branch 'develop' into chore/format-code

tsvetelinak0 commented on CONTRIBUTING.md:
```diff
@@ -1,23 +1,28 @@
 If you are new to [Git](), please take some time to familiarise yourself with Git and Github flow:
+
 - [Git flow](https://guides.github.com/introduction/flow/)
 - [Git offical documentation](https://git-scm.com/)
```
light old typo here..  should be official not offical :)

louiechristie replied:
I updated this in a separate hot-fix commit here: https://github.com/PolyglotDevsLondon/sharespots/commit/083f61b7a291fa483ddc99a5359d5da98463b19e

tsvetelinak0 commented on src/components/index.js:
```diff
@@ -1,7 +1,7 @@
 export default function initComponents() {
-	console.log('****** initComponents() ******');
-	hamburgerComponent();
+  console.log("****** initComponents() ******");
```
js-files are indented with 2 spaces when the indent is set to 4 spaces?

louiechristie replied:
See https://github.com/PolyglotDevsLondon/sharespots/pull/126#discussion_r579161539

tsvetelinak0 replied:
👌

tsvetelinak0 commented on src/styles/base/_variables.scss:
```diff
@@ -48,64 +50,64 @@ $h5-heading-font: $heading-font !default;
-    md: 16,
-    lg: 16,
-    xl: 18,
+  base: 16,
```
I know this is old but would like to ask here when we specify the root variables I think we are missing the units, e.g. px, em, rem, %? otherwise we have only numbers which I guess the browser doesn't apply correctly?

louiechristie replied:
@tsvetelinak0 I think you are right. I searched and looks like $base-font-size isn't actually being used. Can you create a new Trello ticket for updating this, and for also doing breakpoints as CSS variables and generally auditing that css variables are being used correctly?

tsvetelinak0 replied:
ace! sure will keep an eye!

tsvetelinak0 commented on src/styles/components/_amenity_rating.scss:
```diff
@@ -3,43 +3,43 @@
-    }
+  max-width: 120px;
+
+  &-icon {
```
...and the scss files are going from 4 to 2 indentation spaces? Lead dev have the final word, as far as I remember we agreed to 4 spaces at the beginning? need to check with @lili2311 ;)

louiechristie replied:
Think we discussed this in the last meeting and Lilly said 2 spaces was fine. The more important thing is to be consistent, automate it, and not to spend too much time arguing over 2/4 spaces.

tsvetelinak0 replied:
💯

tsvetelinak0 reviewed (changes requested):
left some notes to be clarified..in the meantime agree with the suggestions and approve the changes!

louiechristie committed:
chore: allow commit .vscode workspace settings files

Feedback from tech lead was to allow comiting .vscide workspace settings files
This because workspace settings are different to user settings.
(User settings stay on the individual developer's computer)

louiechristie committed:
Merge branch 'master' into chore/format-code

louiechristie committed:
Merge branch 'develop' into chore/format-code

louiechristie committed:
Merge branch 'develop' into chore/format-code

louiechristie committed:
ci: autoformat js, scss, md pre commit

louiechristie committed:
fix: allow commit .vscode workspace settings files

louiechristie commented:
> left some notes to be clarified..

@tsvetelinak0 I've made some updates and left some replies to your comments. Please can you click on 'Resolve Conversation' if you are happy with my replies.

Please can you review again and approve?

tsvetelinak0 reviewed (approved):
looks great! neat and tidy! good to go! 💯
